### PR TITLE
Feat/16634/buttonicon housekeeping

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Housekeeping 🧹 items for `ButtonIcon`

## Screenshots/Screencaps

No change in UI

## Manual Testing Steps

`yarn test:unit:jest .ui/components/component-library/button-icon/button-icon.test.js`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

## Housekeeping Checklist 

Ensure that `ButtonIcon` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### CODE REVIEW

File: ui/components/component-library/button-icon/button-icon.js [hunk 1]
```diff
@@ -84,14 +85,14 @@ ButtonIcon.propTypes = {
   /**
    * The name of the icon to display. Should be one of ICON_NAMES
    */
-  icon: PropTypes.string.isRequired, // Can't set PropTypes.oneOf(ICON_NAMES) because ICON_NAMES is an environment variable
+  iconName: PropTypes.oneOf(ICON_NAMES),
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-icon/button-icon.js:88]: This works now? And should it still be required?

File: ui/components/component-library/button-icon/README.mdx [hunk 2]
```diff
@@ -15,19 +15,19 @@ The `ButtonIcon` accepts all props below as well as all [Box](/docs/ui-component
 
 <ArgsTable of={ButtonIcon} />
 
-### Icon<span style={{ color: 'red' }}>\*</span>
+### Icon Name<span style={{ color: 'red' }}>\*</span>
 
 Use the required `icon` prop with `ICON_NAMES` object from `./ui/components/component-library/icon` to select icon.
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-icon/README.mdx:20]: Update prop name
```suggestion
Use the required `iconName` prop with `ICON_NAMES` object from `./ui/components/component-library/icon` to select icon.
```

File: ui/components/component-library/button-icon/button-icon.stories.js [hunk 3]
```diff
@@ -60,7 +60,7 @@ export default {
     href: {
       control: 'string',
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-icon/button-icon.stories.js:61]: Update to correct control type
```suggestion
      control: 'text',
```

File: ui/components/component-library/button-icon/README.mdx (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/README.mdx:20]: We could also add this to help engineers fine the icon they need
```
Use the [IconSearch](/story/ui-components-component-library-icon-icon-stories-js--default-story) story to find the icon you want to use.
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left some suggestions. Also some failing accessibility tests

Screenshot 2022-11-23 at 12 50 14 PM
Screenshot 2022-11-23 at 12 50 29 PM
Screenshot 2022-11-23 at 12 50 44 PM
Screenshot 2022-11-23 at 12 50 59 PM

**Review by georgewrmarshall — APPROVED** [review]
LGTM

**Review by NidhiKJha — APPROVED** [review]

File: ui/components/component-library/button-icon/button-icon.js [hunk 4]
```diff
@@ -84,14 +85,14 @@ ButtonIcon.propTypes = {
   /**
    * The name of the icon to display. Should be one of ICON_NAMES
    */
-  icon: PropTypes.string.isRequired, // Can't set PropTypes.oneOf(ICON_NAMES) because ICON_NAMES is an environment variable
+  iconName: PropTypes.oneOf(ICON_NAMES).isRequired,
```
Comment (Gudahtt) [inline ui/components/component-library/button-icon/button-icon.js:88]: This is causing a console error during certain e2e tests. I saw it locally with the test `Swaps - notifications -> AssertionError [ERR_ASSERTION]: Expected values to be strictly equal`.

The error is:

```
Warning: Invalid argument supplied to oneOf, expected an array
```

`ICON_NAMES` is an object here, a mapping of snake-case icon names to kebab-case. I'm guessing you meant to something like this here:

```
iconName: PropTypes.oneOf(Object.values(ICON_NAMES)).isRequired,
```

**Pull request merged**